src/dataflow/src/sink/kafka.rs

```diff
@@ -612,6 +619,17 @@ where
             }
         }
 
+        // If we don't have ready rows, our write frontier equals the minimum
+        // of the input frontier and any stashed timestamps.
+        // While we still have ready rows that we're emitting, hold the write
+        // frontier at the previous time.
+        if ready_rows.is_empty() {
+            write_frontier.borrow_mut().clear();
+            let mut write_frontier = write_frontier.borrow_mut();
+            write_frontier.extend(pending_rows.keys().cloned());
```

Comment:
I think we don't need this, because any timestamps from `pending_rows` that would be less than the input frontier would no longer be in `pending_rows`.

Reply:
I'm not 100% sure, because data goes from pending -> ready when its ahead of the input frontier and the durability frontier

Reply:
I think min of {pending, input_frontiers} is correct but I need to think about it more

Reply:
Ah true, I forgot about the  durability frontier for a moment.